app/src/src/components/modals/OptionModal.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Text, View, TouchableHighlight } from 'react-native';
+import { Picker } from '@react-native-picker/picker';
+
+// common selection inside modals
+const OptionModal = ({
+  visible,
+  setVisible,
+  title,
+  type,
+  setType,
+  choices,
+  setZero,
+  styles,
+}) => {
+  const onPickerChange = (itemValue, _itemIndex) => {
+    let zero = 999;
+    if (itemValue == 0) zero = 0;
+    setType(itemValue);
+    setZero(zero);
+  };
+
+  const getOptions = () => {
+    let temp = [];
+    if (choices.length) {
+      for (let i = 0; i < choices.length; i++) {
+        temp.push(
+          <Picker.Item
+            key={i}
+            label={choices[i]}
+            value={i}
+            style={styles.modalOptions}
+          />
+        );
+      }
+    }
+    return temp;
```

Comment:
Since we are just iterating over the choices, we can use the map function instead.
`return choices.map((choice) => <Picker.Item ..../>);`.